### ISSUE TITLE
After changing the size of the container, it starts to poorly scroll

### ISSUE THREAD

**Issue by Fokir** [rintoj/angular2-virtual-scroll#238]
[Youtube video](https://www.youtube.com/watch?v=HrHVTotQnA4&feature=youtu.be)
When resizing with "parentScroll" it starts behaving strangely. Jumps when scrolling, apparently it does not calculate the height of the elements when changing sizes

**Comment by speige** [comment]
It looks like you're doing something we don't officially support: Changing the size of an item after it has been rendered.

This should be easy to fix. I'll code up something & submit a new version to npm.
It'll be a method to invalidate the cached measurements.

In the meantime, you can try one of these 3 temporary workarounds:

//1) might work, but will work better after I code up a hotfix
```
[enableUnequalChildrenSizes]="false"
```

//2) should work
```
public alwaysFalse(item1, item2) {
   return false;
}
public forceRemeasure() {
    this.myItems = this.myItems.splice(0); // clone array
}
```
```
(compareItems)="alwaysFalse"
[items]="myItems"
```

//3) should work
```
//variable needs to change value when items are resized. value should be exact pixels of rendered item
[childWidth]="myWidth" [childHeight]="myHeight"
```

Here's the reasoning:

In your video it looks like your items are all the same size, However, it looks like when you open the sidebar all of the items shrink at the same time, but they're still equal size to each other. 
[enableUnequalChildrenSizes]="false" is the correct setting here. This setting only needs only to be true if each item has a different size compared to the others items. 

enableUnequalChildrenSizes causes a performance hit because it has to measure every single item, instead of only one.

The root issue is the scrollbar can be jumpy if it has incorrect measurements. Once an item has been measured it is cached & won't be re-measured.

**Comment by speige** [comment]
I've committed a fix for this. It's not yet available on npm, but will be soon.

After release, I think your best option would be 
[enableUnequalChildrenSizes]="false"
After elements change size due to side-bar opening, call this method: virtualScroll.invalidateAllCachedMeasurements();

**Comment by speige** [comment]
This is now available on npm as version 0.4.13
Please re-open if this doesn't solve your issue.
